src/lock.c

```diff
@@ -20,7 +20,12 @@ SDB_API const char *sdb_lock_file(const char *f) {
 	return buf;
 }
 
+#if __SDB_WINDOWS__ && !__CYGWIN__
```

Comment:
Just FYI, in the future I think we should move to different kind of pre-compiler checks. It's easier to let `meson` check if a function (e.g. `getpid`) is available and then define a `HAVE_GETPID=1` if it is. Based on that variable you can then define `os_getpid` in the proper way.

By doing this way we don't need to know and list all possible systems that support one function, we rely directly on what the currently used compiler is able to compile.

Reply:
Sgtm